lib/widget/lottie/native/lottiestate.dart

```diff
@@ -8,65 +9,139 @@ import 'package:ensemble/widget/widget_util.dart';
 import 'package:flutter/widgets.dart';
 import 'package:lottie/lottie.dart';
 
-class LottieState extends WidgetState<EnsembleLottie> {
+class LottieState extends WidgetState<EnsembleLottie>
+    with SingleTickerProviderStateMixin {
+  @override
+  void initState() {
+    super.initState();
+
+    final controller = widget.controller;
+    controller.lottieController = AnimationController(vsync: this);
+
+    final animationStatusActionMap = <AnimationStatus, EnsembleAction?>{
+      AnimationStatus.forward: controller.onForward,
+      AnimationStatus.reverse: controller.onReverse,
+      AnimationStatus.dismissed: controller.onStop,
+      AnimationStatus.completed: controller.onComplete,
+    };
+
+    controller.lottieController!.addStatusListener(
+      (status) {
+        if (animationStatusActionMap[status] != null) {
+          ScreenController().executeAction(
```

Comment:
this code that executes the action should be inside LottieController. That's what the controller is for. Also that way you can re-use most of the code between web and native